presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -346,14 +370,37 @@ protected String generateTemporaryTableName()
     @Override
     public void commitCreateTable(JdbcOutputTableHandle handle)
     {
-        StringBuilder sql = new StringBuilder()
-                .append("ALTER TABLE ")
-                .append(quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTemporaryTableName()))
-                .append(" RENAME TO ")
-                .append(quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTableName()));
+        renameTable(
+                handle.getCatalogName(),
+                new SchemaTableName(handle.getSchemaName(), handle.getTemporaryTableName()),
+                new SchemaTableName(handle.getSchemaName(), handle.getTableName()));
+    }
 
-        try (Connection connection = getConnection(handle)) {
-            execute(connection, sql.toString());
+    @Override
+    public void renameTable(JdbcTableHandle handle, SchemaTableName newTable)
+    {
+        renameTable(null, handle.getSchemaTableName(), newTable);
```

Comment:
Why are we passing `null` rather than `handle.getCatalogName()`?

Reply:
I did this because the MySQL driver specifically doesn't support qualifying the alter table statement with the catalog.  Because this isn't obvious, I instead overrode the `renameTable` method in `MySqlClient`, where we call the super method except with a nulled out catalog name (and a comment explaining why this is done).